src/main.cpp

```diff
@@ -1,151 +1,119 @@
-#include <Arduino.h>
+#include <stdio.h>
+#include <string.h>
 
+#include "hx711.h"
+#include "hx711_noblock.pio.h"
+#include "pico/stdlib.h"
 #include "shoot.h"
 #include "utils.h"
 
-void setup()
-{
-    Serial.begin(115200);
+void update_signal(int &key_input) {
+    // l (led)
+    if (key_input == 108) {
+        utils::flash_led(LED_BUILTIN);
+    }
 
-    // Flash LED
-    utils::flash_led(LED_BUILTIN);
+    // b - beep
+    if (key_input == 98) {
+        shoot::throttle_code = 1;
+        shoot::telemetry = 1;
+    }
 
-    // pwm config
-    // Note that PWM needs to be setup first, 
-    // because the dma dreq requires tts::pwm_slice_num
-    tts::pwm_setup();
+    // r - rise
+    if (key_input == 114) {
+        if (shoot::throttle_code >= ZERO_THROTTLE and
+            shoot::throttle_code <= MAX_THROTTLE) {
+            // Check for max throttle
+            if (shoot::throttle_code == MAX_THROTTLE) {
+                printf("Max Throttle reached\n");
+            } else {
+                shoot::throttle_code += 50;
+                if (shoot::throttle_code > MAX_THROTTLE)
+                    shoot::throttle_code = MAX_THROTTLE;
+                printf("Throttle: %i\n", shoot::throttle_code - ZERO_THROTTLE);
+            }
+        } else {
+            printf("Motor is not in throttle mode\n");
+        }
+    }
 
-    // dma config
-    tts::dma_setup();
+    // f - fall
+    if (key_input == 102) {
+        if (shoot::throttle_code <= MAX_THROTTLE &&
+            shoot::throttle_code >= ZERO_THROTTLE) {
+            if (shoot::throttle_code == ZERO_THROTTLE) {
+                printf("Throttle is zero\n");
+            } else {
+                shoot::throttle_code -= 50;
+                printf("Throttle: %i\n", shoot::throttle_code - ZERO_THROTTLE);
+            }
+        } else {
+            printf("Motor is not in throttle mode\n");
+        }
+    }
 
-    // Set repeating timer
-    // NOTE: this can be put in main loop to start 
-    // repeating timer on key press (e.g. a for arm)
-    shoot::rt_setup();
+    // q - disarm
```

Comment:
zero throttle is *not* disarm. Disarming a motor means stop sending dshot commands. A "silent" disarm could be to just send the `0` command. (Note that the motor will start beeping after some time). Please choose what this is meant to do and rename accordingly